AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginAuthCognitoTests/AWSDataStoreAuthBaseTest.swift

```diff
@@ -379,40 +335,42 @@ extension AWSDataStoreAuthBaseTest {
             .sink { event in
                 // subscription fulfilled
                 if event.eventName == dataStoreEvents.subscriptionsEstablished {
-                    expectations.subscriptionsEstablished.fulfill()
+                    Task { await expectations.subscriptionsEstablished.fulfill() }
                 }
-
                 // modelsSynced fulfilled
                 if event.eventName == dataStoreEvents.modelSynced {
                     modelSyncedCount += 1
                     if modelSyncedCount == expectedModelSynced {
-                        expectations.modelsSynced.fulfill()
+                    Task { await expectations.modelsSynced.fulfill() }
                     }
                 }
 
                 if event.eventName == dataStoreEvents.ready {
-                    expectations.ready.fulfill()
+                    Task { await expectations.ready.fulfill() }
                 }
             }
             .store(in: &requests)
 
-        Amplify.DataStore.start { _ in }
-
-        wait(for: [expectations.subscriptionsEstablished,
-                   expectations.modelsSynced,
-                   expectations.ready],
-             timeout: 60)
-
+        try await Amplify.DataStore.start ()
+        
+        await waitForExpectations([expectations.subscriptionsEstablished,
+                                   expectations.modelsSynced,
+                                   expectations.ready],
+                             timeout: 60)
+    }
+    
+    enum crudError: Error {
+        case notSaved(Error: String?)
+        case notDeleted(Error: String?)
     }
```

Comment:
what's this?

Reply:
removed it: thought of catching error-specific failures in the catch of the do-catch block. got clarified with 
`} catch let error as DataStoreError {`